packages/compartment-mapper/test/map-node-modules.test.js

```diff
@@ -0,0 +1,63 @@
+import 'ses';
+
+import fs from 'node:fs';
+import url from 'node:url';
+import test from 'ava';
+import { mapNodeModules } from '../src/node-modules.js';
+import { makeReadPowers } from '../src/node-powers.js';
+
+/**
+ * @import {CompartmentDescriptor} from '../src/types.js'
+ */
+
+// since we cannot dynamically define tests in AVA (🤦‍♂️), this test will run in a loop.
+test(`mapNodeModules() should return compartment descriptors containing shortest path`, async t => {
+  await null;
+  t.timeout(5000);
+  /** @type {string[] | undefined} */
+  let expectedPath;
+  const readPowers = makeReadPowers({ fs, url });
+  const { maybeRead } = readPowers;
+
+  // inserts a random delay before the read
+  readPowers.maybeRead = async specifier => {
+    return new Promise(resolve => {
+      const delay = Math.random() * 50;
+      return setTimeout(resolve, delay);
+    }).then(() => maybeRead(specifier));
+  };
```

Comment:
As a matter of style, you’ve got some options that would simplify this. The `async` is unnecessary if you return a `new Promise` explicitly, and you can also boil it down to an arrow without return or braces.

But, on the other hand, you could factor out a `delay(ms) => Promise` utility function, keeping the `async`, like

```js
async specifier => {
  await delay(ms);
  return maybeRead(specifier);
};
```

Reply:
I've changed this. 😈